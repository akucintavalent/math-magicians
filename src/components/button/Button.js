import React from 'react';
import PropTypes from 'prop-types';
import './button.css';

class Button extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { buttonName, handleClick, isOrange } = this.props;
    return (
      <button type="button" onClick={handleClick} name={buttonName} className={isOrange ? 'Operations' : ''}>{buttonName}</button>
    );
  }
}

Button.propTypes = {
  buttonName: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  isOrange: PropTypes.bool.isRequired,
};

export default Button;
