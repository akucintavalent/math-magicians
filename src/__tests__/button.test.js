import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import Button from '../components/button/Button';

describe('test text button', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<Button
        key="AC"
        handleClick={() => { }}
        buttonName="AC"
        isOrange={false}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('test click ', async () => {
    const handleClick = jest.fn();
    render(<Button
      key="AC"
      handleClick={handleClick}
      buttonName="AC"
      isOrange={false}
    />);
    await userEvent.click(screen.getByRole('button'));
    expect(handleClick).toHaveBeenCalled();
  });
});
