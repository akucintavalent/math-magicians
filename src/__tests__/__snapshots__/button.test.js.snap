// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className=""
  name="AC"
  onClick={[Function]}
  type="button"
>
  AC
</button>
`;

exports[`test text button renders correctly 1`] = `
<button
  className=""
  name="AC"
  onClick={[Function]}
  type="button"
>
  AC
</button>
`;
